BorrowBookControl.java

```diff
@@ -2,116 +2,111 @@
 import java.util.List;
 
 public class BorrowBookControl {
-	
-	private BorrowBookUI UI;
-	
-	private library LIBRARY;
-	private member M;
-	private enum CONTROL_STATE { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
-	private CONTROL_STATE State;
-	
-	private List<book> PENDING;
-	private List<loan> COMPLETED;
-	private book BOOK;
-	
-	
+
+	private BorrowBookUI borrowBookUi;
+	private library library1;
+	private member member;
+
+	private enum BookControlState {
+		INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED
+	};
+
+	private BookControlState borrowBookState;
+
+	private List<book> pendingList;
+	private List<loan> completedList;
+	private book book;
+
 	public BorrowBookControl() {
-		this.LIBRARY = LIBRARY.INSTANCE();
-		State = CONTROL_STATE.INITIALISED;
+		this.library1 = library1.instanceLibrary();
+		borrowBookState = BookControlState.INITIALISED;
 	}
-	
 
-	public void setUI(BorrowBookUI ui) {
-		if (!State.equals(CONTROL_STATE.INITIALISED)) 
-			throw new RuntimeException("BorrowBookControl: cannot call setUI except in INITIALISED state");
-			
-		this.UI = ui;
-		ui.Set_State(BorrowBookUI.UI_STATE.READY);
-		State = CONTROL_STATE.READY;		
+	public void setBorrowBookUi(BorrowBookUI newUi) {
+		if (!borrowBookState.equals(BookControlState.INITIALISED)) {
+			throw new RuntimeException("BorrowBookControl: cannot call setBorrowBookUi except in INITIALISED state");
+		}
+		this.borrowBookUi = newUi;
+		newUi.setBorrowBookUiState(BorrowBookUI.BorrowBookUiState.READY);
+		borrowBookState = BookControlState.READY;
 	}
 
-		
-	public void Swiped(int MEMMER_ID) {
-		if (!State.equals(CONTROL_STATE.READY)) 
+	public void memberCardSwiped(int memberId) {
+		if (!borrowBookState.equals(BookControlState.READY)) {
 			throw new RuntimeException("BorrowBookControl: cannot call cardSwiped except in READY state");
-			
-		M = LIBRARY.MEMBER(MEMMER_ID);
-		if (M == null) {
-			UI.Display("Invalid memberId");
+		}
+		member = library1.getMember(memberId);
+		if (member == null) {
+			borrowBookUi.Display("Invalid memberId");
 			return;
 		}
-		if (LIBRARY.MEMBER_CAN_BORROW(M)) {
-			PENDING = new ArrayList<>();
-			UI.Set_State(BorrowBookUI.UI_STATE.SCANNING);
-			State = CONTROL_STATE.SCANNING; }
-		else 
-		{
-			UI.Display("Member cannot borrow at this time");
-			UI.Set_State(BorrowBookUI.UI_STATE.RESTRICTED); }}
-	
-	
-	public void Scanned(int bookId) {
-		BOOK = null;
-		if (!State.equals(CONTROL_STATE.SCANNING)) {
+		if (library1.memberCanBorrow(member)) {
+			pendingList = new ArrayList<>();
+			borrowBookUi.setBorrowBookUiState(BorrowBookUI.BorrowBookUiState.SCANNING);
+			borrowBookState = BookControlState.SCANNING;
+		} else {
+			borrowBookUi.Display("Member cannot borrow at this time");
+			borrowBookUi.setBorrowBookUiState(BorrowBookUI.BorrowBookUiState.RESTRICTED);
+		}
+	}
+
+	public void scanneBook(int bookId) {

```

Comment:
@GoJoeyGo scanneBook into scanBook just a simple spelling mistake.
Thanks

Reply:
happy to change thank you for noticing

Reply:
no worries @GoJoeyGo